great_expectations/data_context/store/store_backend.py

```diff
@@ -18,14 +20,30 @@ class StoreBackend(metaclass=ABCMeta):
     """
 
     IGNORED_FILES = [".ipynb_checkpoints"]
+    STORE_ID_KEY = (".ge_store_id",)
+    STORE_ID_PREFIX = "store_id = "
 
-    def __init__(self, fixed_length_key=False):
+    def __init__(self, fixed_length_key=False, store_id: Optional[uuid.UUID] = None):
```

Comment:
It looks to me like this parameter is being ignored; shouldn't we set the key to this value if it's provided? It also seems to me that we should accept both the string and the UUID version of the uuid here.

Reply:
A few questions after thinking about this and trying to implement:
1. I can't think of a case where we want to manually set the `store_id` rather than have it be generated for the user on instantiation or read from an existing store. We also want to make sure it is anonymized. Is it reasonable to remove this parameter? 
2. If NO to question 1: should a `store_id` set via this parameter as a `str` be validated to make sure it is a valid uuid4, or is any string sufficient in this case? I'm assuming we want to make sure it is anonymized, so probably validate.

Reply:
Good point. I think just remove the param.

Reply:
FYI - I'm also adding in automatic initialization of the store_backend_id in the various StoreBackend implementations when the StoreBackend is instantiated (rather than waiting for the property to be called). I think we want that feature but please let me know if that's a bad assumption!